pkg/utils/patch_test.go

```diff
@@ -0,0 +1,194 @@
+					Error:                          nil,
+				},
+			},
+			updatedSnapshot: &crdv1.VolumeSnapshot{
+				Status: &crdv1.VolumeSnapshotStatus{
+					BoundVolumeSnapshotContentName: nil,
+					CreationTime:                   nil,
+					ReadyToUse:                     nil,
+					RestoreSize:                    nil,
+					Error:                          nil,
+				},
+			},
+			expectedPatch: `{"status":{"boundVolumeSnapshotContentName":null}}`,
+		},
+		{
+			name: "1-4: patch snapshotclass",
+
+			originalSnapshot: &crdv1.VolumeSnapshot{
+				Spec: crdv1.VolumeSnapshotSpec{
+					VolumeSnapshotClassName: stringPtr("old"),
+				},
+			},
+			updatedSnapshot: &crdv1.VolumeSnapshot{
+				Spec: crdv1.VolumeSnapshotSpec{
+					VolumeSnapshotClassName: stringPtr("test"),
+				},
+			},
+			expectedPatch: `{"spec":{"volumeSnapshotClassName":"test"}}`,
+		},
+		{
+			name: "1-5: patch snapshotclass no change",
+			originalSnapshot: &crdv1.VolumeSnapshot{
+				Spec: crdv1.VolumeSnapshotSpec{
+					VolumeSnapshotClassName: stringPtr("test"),
+				},
+			},
+			updatedSnapshot: &crdv1.VolumeSnapshot{
+				Spec: crdv1.VolumeSnapshotSpec{
+					VolumeSnapshotClassName: stringPtr("test"),
+				},
+			},
+			expectedPatch: `{}`,
+		},
+	}
+	for _, tc := range testcases {
+		t.Logf("test: %v", tc.name)
+
+		patch, err := createSnapshotPatch(tc.originalSnapshot, tc.updatedSnapshot)
+		if err != nil {
+			t.Fatalf("Encountered unexpected error: %v", err)
+		}
+		if string(patch) != tc.expectedPatch {
+			t.Errorf("Patch not equal to expected patch:\n     GOT: %s\nEXPECTED: %s", patch, tc.expectedPatch)
+		}
+	}
+}
+
+func TestCreateSnapshotContentPatch(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		originalSnapshotContent *crdv1.VolumeSnapshotContent
+		updatedSnapshotContent  *crdv1.VolumeSnapshotContent
+
+		expectedPatch string
+	}{
+		{
+			name: "1-1: empty case",
+
+			originalSnapshotContent: &crdv1.VolumeSnapshotContent{},
+			updatedSnapshotContent:  &crdv1.VolumeSnapshotContent{},
+			expectedPatch:           `{}`,
+		},
+		{
+			name: "1-2: patch new status",
+
+			originalSnapshotContent: &crdv1.VolumeSnapshotContent{},
+			updatedSnapshotContent: &crdv1.VolumeSnapshotContent{
+				Status: &crdv1.VolumeSnapshotContentStatus{
+					SnapshotHandle: stringPtr("test"),
+				},
+			},
+			expectedPatch: `{"status":{"snapshotHandle":"test"}}`,
+		},
+		{
+			name: "1-3: clear status",
+
+			originalSnapshotContent: &crdv1.VolumeSnapshotContent{
+				Status: &crdv1.VolumeSnapshotContentStatus{
+					SnapshotHandle: stringPtr("test"),
+					CreationTime:   nil,
+					ReadyToUse:     nil,
+					RestoreSize:    nil,
+					Error:          nil,
+				},
+			},
+			updatedSnapshotContent: &crdv1.VolumeSnapshotContent{
+				Status: &crdv1.VolumeSnapshotContentStatus{
+					SnapshotHandle: nil,
+					CreationTime:   nil,
+					ReadyToUse:     nil,
+					RestoreSize:    nil,
+					Error:          nil,
+				},
```

Comment:
Add some tests where the updated status has non-nil CreationTime, ReadyToUse, RestoreSize.

Reply:
Added these tests